src/test/java/net/openhft/chronicle/core/UnsafeMemoryTest.java

```diff
@@ -79,102 +93,74 @@ public void readShort() {
     }
 
     @Test
-    public void writeInt() {
+    public void readWriteInt() {
         for (int i = 0; i <= 64; i++)
             if (onHeap == null) {
-                memory.writeInt(addr + i, 0);
+                memory.writeInt(addr + i, INT_VAL);
+                assertEquals(INT_VAL, memory.readInt(addr + i));
             } else {
-                memory.writeInt(object, addr + i, 0);
+                memory.writeInt(object, addr + i, INT_VAL);
+                assertEquals(INT_VAL, memory.readInt(object, addr + i));
             }
     }
 
     @Test
     public void writeOrderedInt() {
         for (int i = 0; i <= 64; i++)
             if (onHeap == null) {
-                memory.writeOrderedInt(addr + i, 0);
-            } else {
-                memory.writeOrderedInt(object, addr + i, 0);
-            }
-    }
-
-    @Test
-    public void readInt() {
```

Comment:
why remove?

Reply:
Because the read and write tests are merged and actually asserts operations produce the correct effect.